fix comparison: integer MYSQL_VERSION_ID with string MARIADB_BASE_VERSION
Author: Samuel Moors, Vrije Universiteit Brussel (VUB)
diff -ur DBD-mysql-4.048.orig/dbdimp.c DBD-mysql-4.048/dbdimp.c
--- DBD-mysql-4.048.orig/dbdimp.c	2018-09-10 21:35:12.000000000 +0200
+++ DBD-mysql-4.048/dbdimp.c	2018-10-10 11:51:00.138239313 +0200
@@ -1907,14 +1907,14 @@
                          (SvTRUE(*svp) ? "utf8" : "latin1"));
         }
 
-#if (MYSQL_VERSION_ID >= 50723) && (MYSQL_VERSION_ID < MARIADB_BASE_VERSION)
+#if !defined(MARIADB_BASE_VERSION) && (MYSQL_VERSION_ID >= 50723)
         if ((svp = hv_fetch(hv, "mysql_get_server_pubkey", 23, FALSE)) && *svp && SvTRUE(*svp)) {
           my_bool server_get_pubkey = 1;
           mysql_options(sock, MYSQL_OPT_GET_SERVER_PUBLIC_KEY, &server_get_pubkey);
         }
 #endif
 
-#if (MYSQL_VERSION_ID >= 50600) && (MYSQL_VERSION_ID < MARIADB_BASE_VERSION)
+#if !defined(MARIADB_BASE_VERSION) && (MYSQL_VERSION_ID >= 50600)
         if ((svp = hv_fetch(hv, "mysql_server_pubkey", 19, FALSE)) && *svp) {
           STRLEN plen;
           char *server_pubkey = SvPV(*svp, plen);
